cdap-watchdog/src/main/java/co/cask/cdap/logging/appender/kafka/KafkaLogAppender.java

```diff
@@ -206,14 +206,23 @@ private void publishMessages(List<KeyedMessage<String, byte[]>> buffer,
 
       if (blockForMessage) {
         blockingThread = Thread.currentThread();
+        LogMessage message = null;
         try {
           if (isRunning()) {
-            buffer.add(createKeyedMessage(messageQueue.take()));
+            message = messageQueue.take();
+            buffer.add(createKeyedMessage(message));
             maxBufferSize--;
           }
         } catch (InterruptedException e) {
           // just ignore and keep going. This happen when this publisher is getting shutdown, but we still want
           // to publish all pending messages.
+        } catch (Throwable t) {
+          if (message == null) {
+            contextAware.addError("Got exception while taking message from queue", t);
```

Comment:
This is to catch exceptions from `loggingEventSerializer.toBytes(logMessage)` in `createKeyedMessage`